### PR TITLE
JSDK-2549 add trackSwitchOffMode property

### PR DESCRIPTION
**Pull request by makarandp0**
This change adds `trackSwitchOffMode` property in bandwidthProfile.video object - 
spec:  https://code.hq.twilio.com/client/video-sdk-api/pull/55

Besides the spec it also moves bandwidthprofile tests from connect.js over to its file. 

TODO: 
- [x] verify test runs after VMS and [RSP](https://issues.corp.twilio.com/browse/VIDEO-2332) changes are deployed in stage ( )
- [x] remove debug code (console.logs)
 
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: lib/signaling/v2/twilioconnectiontransport.js [hunk 1]
```diff
@@ -234,6 +234,8 @@ class TwilioConnectionTransport extends StateMachine {
         message.bandwidth_profile = createBandwidthProfilePayload(
           this._bandwidthProfile);
       }
+      // eslint-disable-next-line no-console
+      console.log('message.bandwidth_profile:', message.bandwidth_profile);
```
Comment (syerrapragada) [inline lib/signaling/v2/twilioconnectiontransport.js]: NitPick: Remove console log

File: test/integration/spec/bandwidthprofile.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+          x => `and the publish priority of the Passive Speaker's LocalVideoTrack is "${x}"`
+        ]
+      ], ([trackLimitOptions, dominantSpeakerPriority, dominantSpeakerPublishPriority, passiveSpeakerPublishPriority]) => {
+        const priorityRanks = {
+          [PRIORITY_HIGH]: 1,
+          [PRIORITY_STANDARD]: 2,
+          [PRIORITY_LOW]: 3
+        };
+
+        // NOTE(mmalavalli): Since "dominantSpeakerPriority" only upgrades the publish priority of the Dominant Speaker's
```
Comment (syerrapragada) [inline test/integration/spec/bandwidthprofile.js]: dominantSpeakerPriority is a subscriber-side priority and It should not change publishPriority. isn't it?

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,29 @@ For 1.x changes, go [here](https://github.com/twilio/twilio-video.js/blob/suppor
 
 New Features
 ------------
+- You can now specify the mode to controls track switch off behavior by specifying a
+  property `trackSwitchOffMode` in BandwidthProfile options. This can be set to one of
+  `detected`  - In this mode, RemoteVideoTracks are switched off only when network congestion
```
[the remark this answers is not in the document]
Comment (makarandp0) [reply CHANGELOG.md]: Not sure what you mean - We do have an `enum` for it in globals (https://github.com/twilio/twilio-video.js/blob/a059adeb34ef43e90f7310b2b95c6e70e554bcc3/lib/connect.js#L566)

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,29 @@ For 1.x changes, go [here](https://github.com/twilio/twilio-video.js/blob/suppor
 
 New Features
 ------------
```
[the remark this answers is not in the document]
Comment (makarandp0) [reply CHANGELOG.md]: Thanks @anna-vasilko, @aymenn  can you please take a look at this changelog as well.

File: lib/connect.js [hunk 5]
```diff
@@ -556,6 +558,31 @@ const NetworkQualityVerbosity = {
...
+const TrackSwitchOffMode = {
+  /**
+   * In this mode, {@link RemoteVideoTrack}s are switched off only when network congestion
+   * is detected.
+   */
+  detected: 'detected',
+
+  /**
+   * In this mode, {@link RemoteVideoTrack}s are pro-actively switched off when network
+   * congestion is predicted by the bandwidth estimation mechanism.
```
[the remark this answers is not in the document]
Comment (makarandp0) [reply lib/connect.js]: The difference is `detected` v/s `predicted`
 'detected' - is when network congestion actually happens.
'predicted' - is where VMS predicts that network congestion is going to happen.

File: test/integration/spec/bandwidthprofile.js (hunk 2 above)
Comment (makarandp0) [reply test/integration/spec/bandwidthprofile.js]: yes, it does not change the publish side priority. I will edit the comment.

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (syerrapragada) [reply CHANGELOG.md]: I mean in the changelog format could we use a list to describe modes:

- `detected` - .....
- `predicted` - ...
- `disabled` - ...

**Pull request merged**